tests/test_temporal.py

```diff
@@ -2356,3 +2377,754 @@ async def test_beta_graph_parallel_execution_in_workflow(client: Client):
+        result = await multi_model_error_test_agent.run(prompt, model=test_model_error_unregistered)
+        return result.output  # pragma: no cover
+
+
+@workflow.defn
+class MultiModelWorkflowSelectModel2String:
+    @workflow.run
+    async def run(self, prompt: str) -> str:
+        result = await multi_model_string_test_agent.run(prompt, model='second_model')
+        return result.output
+
+
+async def test_temporal_agent_multi_model_registration():
+    """Test that TemporalAgent correctly registers multiple models with unique activity names."""
+    test_model1 = TestModel(custom_output_text='Model 1 response')
+    test_model2 = TestModel(custom_output_text='Model 2 response')
+    test_model3 = TestModel(custom_output_text='Model 3 response')
+
+    agent = Agent(test_model1, name='multi_model_test')
+    temporal_agent = TemporalAgent(
+        agent,
+        name='multi_model_test',
+        additional_models={'model_2': test_model2, 'model_3': test_model3},
+    )
+
+    assert set(temporal_agent._registered_model_instances.keys()) == {  # pyright: ignore[reportPrivateUsage]
+        'default',
+        'model_2',
+        'model_3',
+    }
+
+    activity_names = {
+        name
+        for activity in temporal_agent.temporal_activities
+        if (name := _temporal_activity_name(activity)) is not None
+    }
+    assert 'agent__multi_model_test__model_request' in activity_names
+    assert 'agent__multi_model_test__model_request_stream' in activity_names
+    assert 'agent__multi_model_test__event_stream_handler' in activity_names
+    # There should only be a single pair of model activities regardless of registered models.
+    assert len([name for name in activity_names if name.endswith('__model_request')]) == 1
+    assert len([name for name in activity_names if name.endswith('__model_request_stream')]) == 1
+
+
+async def test_temporal_agent_multi_model_reserved_name():
+    """Test that reserved model names raise helpful errors."""
+    test_model1 = TestModel()
+    test_model2 = TestModel()
+
+    agent = Agent(test_model1, name='reserved_name_test')
+    with pytest.raises(UserError, match="Model name 'default' is reserved"):
+        TemporalAgent(
+            agent,
+            name='reserved_name_test',
+            additional_models={'default': test_model2},
+        )
+
+
+async def test_temporal_agent_multi_model_duplicate_name():
+    """Test that duplicate model names (after normalization) raise helpful errors."""
+    test_model1 = TestModel()
+    test_model2 = TestModel()
+    test_model3 = TestModel()
+
+    agent = Agent(test_model1, name='duplicate_name_test')
+    # Keys that normalize to the same value should raise an error
+    with pytest.raises(UserError, match="Duplicate model name 'model2' provided to `additional_models`"):
+        TemporalAgent(
+            agent,
+            name='duplicate_name_test',
+            additional_models={'model2': test_model2, ' model2 ': test_model3},
```

Comment:
Do we actually need to normalize model IDs at all? Except for `default`, I think we can just support any string and leave it up to the user to provide the same one, now that we're no longer using these in activity names, just for dict lookups.